lib/puppet/file_system.rb

```diff
@@ -122,8 +122,8 @@ def self.each_line(path, &block)
   #
   # @api public
   #
-  def self.read(path)
-    @impl.read(assert_path(path))
+  def self.read(path, opts = {})
+    @impl.read(assert_path(path), opts)
```

Comment:
I had something similar to this once. :) It was in line with PUP-927 that I started looking down that route.